### ISSUE TITLE
Increase length of object ids to 128 characters

### ISSUE THREAD

**Issue by AbelHeinsbroek** [OpenWaterAnalytics/EPANET#90]
The current length of object ids is limited to 32 characters. It would be nice to increase this length to 128 characters to allow the usage of UUIDs as object ids.

**Comment by samhatchett** [comment]
good idea ... redefining `MAXID` is easy, but maybe a further step would be to allow null-terminated strings of arbitrary length.

**Comment by eldemet** [comment]
I also like this suggestion, using UUIDs as network ids would a useful feature.

**Comment by AbelHeinsbroek** [comment]
Redefining MAXID is not enough, the id length is encoded in some of the input[n].c files as well

**Comment by samhatchett** [comment]
is there a face-palm emoji?

**Comment by eladsal** [comment]
could be done but might be important to do it in a way which won't break old code (backward compatibility). Also, 128 characters would defiantly break compatibility with EPANET 2.0 GUI, would be good idea to talk about this with the new GUI people.

**Comment by samhatchett** [comment]
good item for the roadmap, anyway. the future may look like migrating to a new set of apis, and adapt current functions to wrap the extended set for compatibility.

**Comment by MarkGrayRESPEC** [comment]
The new GUI we are building does not currently enforce a maximum ID length, it is just treated as an arbitrary Python string.

**Comment by samhatchett** [comment]
I've updated the title of this Issue to be more general. Thinking that the `inp` parser can have its own limits, but no need to impose those on the toolkit user.

**Comment by michaeltryby** [comment]
Since EPANET stores network IDs along with other network data in the fixed length section of the binary output file, this would also affect the layout of binary file and retrieval of model element ID's via the output API.

**Comment by eladsal** [comment]
Moving to 2.3
